### ISSUE TITLE
Cannot correctly suppress warnings on anonymous classes

### ISSUE THREAD

**Issue by rhysr** [phpmd/phpmd#499]
Using phpmd 2.0.16, I'm unable to correctly suppress warnings on anonymous classes

The following outputs
**Avoid unused local variables such as '$unused'.**
```php
<?php

namespace Example;

class Foo 
{
    public function createAnonymousClass()
    {
        new class { 
            /**
             * @SuppressWarnings(PHPMD.UnusedLocalVariable)
             */
            public function bar($unused)
            {
            }
        };  
    }       
}       
```

However, the following examples do not produce any errors

```php
namespace Example;

class Foo 
{
    /** 
     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
     */
    public function createAnonymousClass()
    {   
        new class {
            public function bar($unused)
            {   
            }   
        };  
    }   
}
```
```php
<?php

namespace Example;

class Foo 
{
    public function createAnonymousClass()
    {   
        new class {
            public function bar($unused)
            {   
            }   
        };  
        return $unused;
    }   
}

```

**Comment by ravage84** [comment]
Confirmed:

![unused_local_varaible_suppression_does_not_work_in_anonymous_class](https://user-images.githubusercontent.com/625761/27694632-edf5af4e-5cec-11e7-9c59-71164a40bc91.png)
